legal-auto.py

```diff
@@ -0,0 +1,189 @@
+                package = REQ.post(url).json()
+                # reopen
+                return None
+            if not state in ['acceptable', 'correct', 'unacceptable']:
+                return None
+            if state == 'unacceptable':
+                user = report.get('reviewing_user', None)
+                if not user:
+                    self.message = 'declined'
+                    print self.message
+                    return None
+                comment = report.get('result', None)
+                if comment:
+                    self.message = "@{} declined the legal report with the following comment: {}".format(user, comment)
+                else:
+                    self.message = "@{} declined the legal report".format(user)
+                    print self.message
+                    return None
+                return False
+            #print url, json.dumps(report)
+        self.message = 'ok'
+        return True
+
+    def check_action__default(self, req, a):
+        self.logger.error("unhandled request type %s"%a.type)
+        return True
+
+    def prepare_review(self):
+        url = osc.core.makeurl(self.legaldb, ['requests'])
+        req = REQ.get(url).json()
+        self.open_reviews = {}
+        requests = []
+        for hash in req['requests']:
+            ext_link = str(hash['external_link'])
+            self.open_reviews[ext_link] = list(set(hash['packages']))
+            if ext_link.startswith(self.apinick):
+                rq = ext_link[len(self.apinick):]
+                requests.append('@id='+rq)
+        if len(requests):
+            match = "(state/@name='declined' or state/@name='revoked' or state/@name='superseded')"
+            match += ' and (' + ' or '.join(sorted(requests)) + ')'
+            url = osc.core.makeurl(self.apiurl, ['search', 'request', 'id'], { 'match': match })
+            # prefer POST because of the length
+            root = ET.parse(osc.core.http_POST(url)).getroot()
+            for request in root.findall('request'):
+                self.delete_from_db(request.get('id'))
+
+    def delete_from_db(self, id):
+        url = osc.core.makeurl(self.legaldb, ['requests'], { 'external_link': self.request_nick(id) } )
+        REQ.delete(url)
+
+    # overload as we need to get of the bot_request
+    def _set_review(self, req, state):
+        if self.dryrun:
+            self.logger.debug("dry setting %s to %s with %s"%(req.reqid, state, self.message))
+            return
+
+        self.logger.debug("setting %s to %s"%(req.reqid, state))
+        osc.core.change_review_state(apiurl = self.apiurl,
+                                     reqid = req.reqid, newstate = state,
+                                     by_group=self.review_group,
+                                     by_user=self.review_user, message=self.message)
+        self.delete_from_db(req.reqid)
```

Comment:
This appears to be the only unique line short of dropping functionality from the original. Perhaps call super() and check `self.dryrun` before `delete_from_db()`? The message should be settable via the standard `self.review_messages`.